contracts/pcv/maple/MaplePCVDeposit.sol

```diff
@@ -0,0 +1,153 @@
+    /// @param _mplRewards Maple Rewards contract
+    constructor(
+        address _core,
+        address _pool,
+        address _mplRewards
+    ) CoreRef(_core) {
+        token = IERC20(IPool(_pool).liquidityAsset());
+        /// enforce underlying token is USDC
+        require(
+            address(token) == 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48,
+            "MaplePCVDeposit: Underlying not USDC"
+        );
+        pool = IPool(_pool);
+        mplRewards = IMplRewards(_mplRewards);
+    }
+
+    /// @notice return the amount of funds this contract owns in Maple FDT's
+    /// without accounting for interest earned
+    /// does not account for unrealized losses in the venue
+    function balance() public view override returns (uint256) {
+        return pool.balanceOf(address(this)) / scalingFactor;
+    }
+
+    /// @notice return the underlying token denomination for this deposit
+    function balanceReportedIn() external view returns (address) {
+        return address(token);
+    }
+
+    /// @notice deposit PCV into Maple.
+    /// all deposits are subject to a minimum 90 day lockup,
+    /// no op if 0 token balance
+    /// deposits are then immediately staked to accrue MPL rewards
+    /// only pcv controller can deposit, as this contract would be vulnerable
+    /// to donation / griefing attacks if anyone could call deposit and extend lockup time
+    function deposit() external onlyPCVController {
+        uint256 amount = IERC20(token).balanceOf(address(this));
+        if (amount == 0) {
+            /// no op to prevent wasted gas
+            return;
+        }
+
+        /// pool deposit
+        token.approve(address(pool), amount);
+        pool.deposit(amount);
+
+        /// stake pool FDT for MPL rewards
+        uint256 scaledDepositAmount = amount * scalingFactor;
+        pool.increaseCustodyAllowance(address(mplRewards), scaledDepositAmount);
+        mplRewards.stake(scaledDepositAmount);
+
+        emit Deposit(msg.sender, amount);
+    }
+
+    /// @notice function to start the cooldown process to withdraw
+    /// 1. lp lockup on deposit --> 90 days locked up and can't withdraw
+    /// 2. cool down period, call intend to withdraw -->
+    ///   must wait 10 days before withdraw after calling intend to withdraw function
+    /// 3. after cool down and past the lockup period,
+    ///    have 2 days to withdraw before cool down period restarts.
+    function signalIntentToWithdraw() external onlyPCVController {
+        pool.intendToWithdraw();
+    }
+
+    /// @notice function to cancel a withdraw
+    /// should only be used to allow a transfer when doing a withdrawERC20 call
+    function cancelWithdraw() external onlyPCVController {
+        pool.cancelWithdraw();
+    }
+
+    /// @notice withdraw PCV from Maple, only callable by PCV controller
+    /// @param to destination after funds are withdrawn from venue
+    /// @param amount of PCV to withdraw from the venue
```

Comment:
if losses are sustained in this venue and marked down before exit, they get applied to all users, which means line 124 transfer will fail because it tries to transfer too much out that this contract might not have